src/main/java/manipulateActions/ManipulateAction.java

```diff
@@ -0,0 +1,62 @@
+package manipulateActions;
+
+import fileFilter.IFilter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ManipulateAction implements IManipulateAction {
+    private IManipulateAction dataManipulator;
+    private List<IFilter> manipulatorFilters;
+
+    public ManipulateAction(IManipulateAction dataManipulator) {
+        this.dataManipulator = dataManipulator;
+        this.manipulatorFilters = new ArrayList<IFilter>();
+    }
+
+    @Override
+    public String manipulateData(String fileChunkData) {
+        return dataManipulator.manipulateData(fileChunkData);
+    }
+
+    @Override
+    public boolean isByteRequiredForAction(int data) {
+        return dataManipulator.isByteRequiredForAction(data);
+    }
+
+    @Override
+    public ManipulateActionsType getManipulateActionType() {
+        return dataManipulator.getManipulateActionType();
+    }
+
+    public void addManipulatorFilter(IFilter manipulatorFilter) {
+        this.manipulatorFilters.add(manipulatorFilter);
+    }
+
+    public boolean isAllFiltersActivated() {
+        if (!isActionHaveFilters())
```

Comment:
in the event that `manipulatorFilters` is empty, the `for` loop in the following lines would just not run any amount of times, resulting in a return value of `true`. this is unnecessary